File: doc/codes/rust/examples/api/compute/pattern/set-example.rs
use autd3_rs::geometry::{Autd3, Geometry, offset};
use autd3_rs::units::{m, mm, s};
use autd3_rs::value::{Intensity, Phase};
use autd3_rs_pattern::{add_phase, focus, set_intensity, wavelength};

// HIDE
fn main() {
    // HIDE_END
    let geometry = Geometry::new(vec![Autd3::default()]);

    let mut phases = geometry.phase_buffer();
    let mut intensities = geometry.intensity_buffer();

    set_intensity(Intensity(0x80), &mut intensities);
    focus(
        &geometry,
        geometry.center() + offset(0.0 * mm, 0.0 * mm, 150.0 * mm),
        wavelength(340.0 * m / s),
        &mut phases,
    );
    add_phase(Phase::PI, &mut phases);

    set_intensity(Intensity::MIN, &mut intensities);
    // HIDE
}
// HIDE_END

File: doc/codes/python/examples/api/compute/pattern/set-example.py
import numpy as np
from autd3.geometry import Autd3, Geometry
from autd3.units import m, s
from autd3.value import Intensity, Phase
from autd3_pattern import add_phase, focus, set_intensity, wavelength

geometry = Geometry([Autd3([0.0, 0.0, 0.0], [1.0, 0.0, 0.0, 0.0])])

phases = geometry.phase_buffer()
intensities = geometry.intensity_buffer()

set_intensity(Intensity(0x80), intensities)
focus(
    geometry,
    geometry.center() + np.array([0.0, 0.0, 150.0]),
    wavelength(340 * m / s),
    phases,
)
add_phase(Phase.PI, phases)

set_intensity(Intensity.MIN, intensities)
